src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -12,19 +1,55 @@
+import React from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+interface Props {
+  filter: string;
+  query: string;
+  onFilterChange: (status: string) => void;
+  onSearchChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  onClearSearch: () => void;
+}
+
+export const TodoFilter: React.FC<Props> = ({ filter, query, onFilterChange, onSearchChange, onClearSearch }) => {
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            value={filter}
+            onChange={e => onFilterChange(e.target.value)}
+            data-cy="statusSelect"
+          >
+            <option value="all">All</option>
+            <option value="active">Active</option>
+            <option value="completed">Completed</option>
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={query}
+          onChange={onSearchChange}
+          data-cy="searchInput"
+        />
```

Comment:
Consider adding a label for the input element to improve accessibility. Alternatively, you can use `aria-label` to provide a text description.